Sources/LogInOut.php

```diff
@@ -169,7 +165,7 @@ function Login2()
 		redirectexit();
 
 	// Are you guessing with a script?
-	checkSession();
+	checkSession('login');
```

Comment:
type login didn't exist: https://github.com/SimpleMachines/SMF2.1/blob/e750119e8d604456faa760c7a6f5fdc7050a60f4/Sources/Security.php#L596

Reply:
Correct.   Any value other than put or get.  Simply trying to bypass those edits that are not appropriate.

Reply:
I see no need for this,
otherwise you need to extend the checksession function,
because you use a type which didn't exists -> wrong.

Reply:
Maybe just add a comment at the beginning of checksession to that effect?

The problem is that with no parameter, it assumes you are making a post to the forum.

Reply:
because you do a post to the forum:
![grafik](https://user-images.githubusercontent.com/1782906/31582545-c723b3cc-b186-11e7-9d07-450606126fae.png)

Reply:
The post is of the logon form, not a post of a new message.  Edits like checking for session timeout are intended for posting new messages, and do not make sense during login.

Reply:
We are here in login2, this function is only called at login and this is a post (see my screenshot)

Reply:
What happens if you leave this as `checkSession();`?

Reply:
Assume for a second that we split this PR into two PRs, one that addressed the preg_match issue & this change.  I will answer this change as if this change were made in isolation:  WITHOUT this change, the issues over in #4297 appear - you get the 'session timed out' error.  WITH this change, you get an appropriate 'there is a problem with your cookie' message.  

My reasoning for this change: the first edit in checkSession() compares the session var to the prior value - which will fail, because you were logged off, i.e., you don't have a prior session.  That's why I think those first two checks in checkSession() don't make sense during logon.  

Since fixing the pregmatch pattern solves the problem (the other change in this PR), leaving this at checkSession() actually makes no difference.  The problem is fixed.  But I made this change because, as indicated above, I was believe that those first edits at the top of checkSession() do not apply for logins and may bite us under other circumstances.   

Note that checkSession() is invoked ~200 times throughout the application, under all sets of circumstances (message posts, admin tasks, etc...).   I think the initial edits at the top of checkSession for stale sessions make more sense for message posts, but not for logging in.

Reply:
Thanks for the explanation, @sbulen. I agree with @albertlast's reading of the situation regarding checkSession(). If this PR works without changing `checkSession()` to `checkSession('login')`, I would much prefer to leave it as it was.

Reply:
Done.

Reply:
Thanks